src/_pytest/doctest.py

```diff
@@ -62,6 +62,12 @@ def pytest_addoption(parser):
     parser.addini(
         "doctest_encoding", "encoding used for doctest files", default="utf-8"
     )
+    parser.addini(
+        "doctest_standalone_namespace",
```

Comment:
I can't think of any better options for the name either. 
How about calling it `doctest_context` with a non bool type, or even better, `doctest_reset_context`? That would leave the option up to implement something similar to the plot directive's context. And then, this can remain a bool.

Reply:
I'm OK with `doctest_reset_context`, although perhaps `doctest_empty_context` better conveys what the option actually does?

Reply:
From a technical point of view, "namespace" is a better term than "context". Two function doctests don't interact with each other, which is what "reset context" seems to imply to me. So if you define a variable in one doctest it won't be available in another doctest. 

Rather, they just reuse the module namespace, so if a variable is defined at the module level (not in a doctest), it gets defined automatically in a doctest. Maybe a name like `doctest_no_module_namespace` (although honestly my biggest issue with the current name is that it is too long).

Reply:
IMHO a long option name is OK if it correctly describes the behavior, because it will be written in a `pytest.ini` file only once anyway. 😁

Reply:
I guess my problem isn't so much the length as the fact that the option name seems convoluted. I don't think many people will look at "doctest_standalone_namespace" and understand what it means unless they already know.

Reply:
Fair point, agree.

`doctest_no_module_namespace` seems better in that light then, IMHO.